summingbird-storm/src/main/scala/com/twitter/summingbird/storm/spout/KeyValueSpout.scala

```diff
@@ -67,10 +67,13 @@ class KeyValueSpout[K, V: Semigroup](val in: IRichSpout, summerBuilder: SummerBu
     msgIds.foreach { super.fail(_) }
   }
 
-  def convertToList(msgId: Object): MList[Object] = {
+  /*
+  The msgId Object is a list of individual messageIds of all the aggregated tuples.
+   */
+  def convertToList(msgId: Object): TraversableOnce[AnyRef] = {
```

Comment:
This method has a lot of ways to fail at runtime. Can we make it private and rework?
```scala
private def castToTraverableOnce(shouldBeOptionTO: AnyRef): TraversableOnce[AnyRef] =
  shouldBeOptionTO match {
    case Some(s: TraversableOnce[_]) => s.asInstanceOf[TraversableOnce[AnyRef]]
    case None => Nil // Nil is an object, no allocations, but is traverableonce.
    case otherwise => throw new ClassCastException(s"expected Option[TraversableOnce[AnyRef]] found class: ${otherwise.getClass} with value: $otherwise")
  }
```